src/SDKs/ServiceBus/data-plane/tests/Microsoft.Azure.ServiceBus.Tests/Infrastructure/TaskExtensions.cs

```diff
@@ -0,0 +1,164 @@
+            }
+
+            try
+            {
+                await instance.ConfigureAwait(false);
+            }
+
+            catch (Exception ex)
+            {
+                errorAction?.Invoke(ex);
+            }
+        }
+
+        /// <summary>
+        ///   Executes a task without waiting for or observing its completion.
+        /// </summary>
+        /// 
+        /// <typeparam name="T">The type of return value of the task.</typeparam>
+        /// 
+        /// <param name="instance">The target of this fire-and-forget.</param>
+        /// <param name="errorAction">An action to execute should an exception occur.</param>
+        /// 
+        public static async void FireAndForget<T>(this Task<T> instance, Action<Exception> errorAction = null)
+        {
+            if (instance == null)
+            {
+                return;
+            }
+
+            try
+            {
+                await instance.ConfigureAwait(false);
+            }
+
+            catch (Exception ex)
+            {
+                errorAction?.Invoke(ex);
+            }
+        }
+        
+        /// <summary>
+        ///   Executes a task with a timeout period, ignoring any result should the timeout period elapse.
+        /// </summary>
+        /// 
+        /// <param name="instance">The target of task execution.</param>
+        /// <param name="timeout">The timeout period to allow for the <paramref name="instance"/> to complete.</param>
+        /// <param name="cancellationToken">A cancellation token for signaling the <paramref name="instance" /> that a timeout has occurred and work should be stopped.</param>
+        /// <param name="timeoutAction">An action to take on timeout; if not specified, a <see cref="TimeoutException" /> will be thrown.</param>
+        /// 
+        /// <returns>A task to be resolved on completion or timeout.</returns>
+        /// 
+        public static async Task WithTimeout(this Task instance, TimeSpan timeout, CancellationTokenSource cancellationToken = null, Action timeoutAction = null)
+        {
+            if (instance == null)
+            {
+                throw new ArgumentNullException(nameof(instance));
+            }
+
+            if (instance.IsCompleted || Debugger.IsAttached)
+            {
+                return;
+            }
+
+            using (var timeoutTokenSource = new CancellationTokenSource())
+            {
+                if (instance == await Task.WhenAny(instance, Task.Delay(timeout, timeoutTokenSource.Token)))
+                {
+                    timeoutTokenSource.Cancel();
+                    instance.GetAwaiter().GetResult();
+                    return;
+                }
+            }
+
+            // A timeout occured.  Perform the needed actions to request
+            // cancellation, allow the task to complete unobserved, and to
+            // signal the caller.
+
+            cancellationToken?.Cancel();
+            instance.FireAndForget();
```

Comment:
Why do we need FireAndForget here at all? This task is already running or finished.